### ISSUE TITLE
Triangulated depth with dynamic objects

### ISSUE THREAD

**Issue by VitorGuizilini-TRI** [MARSLab-UMN/DeepMultiviewDepth#4]
Hi, it's me again. I'm using your triangulation for depth estimation from optical flow, and these are some results I am getting on VKITTI2:

GT depth 

![depth](https://user-images.githubusercontent.com/58576956/102646477-f297fe80-4118-11eb-89ca-724696764b13.png)

GT optical flow

![flow](https://user-images.githubusercontent.com/58576956/102646490-f6c41c00-4118-11eb-899c-e038281bbcf8.png)

Triangulate depth

![flow_depth](https://user-images.githubusercontent.com/58576956/102646493-fa57a300-4118-11eb-8271-6a0b52ffcf8d.png)

As you can see, it works fine everywhere except on dynamic objects (and some strange artifacts on the sky). Is that expected, or am I doing something wrong? I'm assuming you don't have that many dynamic objects on the datasets you evaluated on the paper, so that's why I am asking.

**Comment by MathLens** [comment]
Triangulation assumes the points are static across multi-view, so that is expected, which is also one of the reasons we have a depth refinement work based on single view after the triangulation.

**Comment by nauti16** [comment]
> Hi, it's me again. I'm using your triangulation for depth estimation from optical flow, and these are some results I am getting on VKITTI2:
> 
> GT depth
> 
> ![depth](https://user-images.githubusercontent.com/58576956/102646477-f297fe80-4118-11eb-89ca-724696764b13.png)
> 
> GT optical flow
> 
> ![flow](https://user-images.githubusercontent.com/58576956/102646490-f6c41c00-4118-11eb-899c-e038281bbcf8.png)
> 
> Triangulate depth
> 
> ![flow_depth](https://user-images.githubusercontent.com/58576956/102646493-fa57a300-4118-11eb-8271-6a0b52ffcf8d.png)
> 
> As you can see, it works fine everywhere except on dynamic objects (and some strange artifacts on the sky). Is that expected, or am I doing something wrong? I'm assuming you don't have that many dynamic objects on the datasets you evaluated on the paper, so that's why I am asking.

Hi, VitorGuizilini-TRI.
I followed your work using the VKITTI2 dataset, but the depth results I obtained are not as clean as yours. Could you share the method you used?

**Comment by VitorGuizilini-TRI** [comment]
Do you mean my triangulation results, like the one I'm showing above?

**Comment by nauti16** [comment]
Yes, regarding your triangulation result, below is the result I obtained by triangulating the VKITTI2 data. As you can see, the depth result isn't coming out very well.
![image (2)](https://github.com/user-attachments/assets/cb6f9515-d868-4134-97a5-40edde4f4954)


Would any adjustments to the extrinsic parameters or other preprocessing steps be necessary?

**Comment by VitorGuizilini-TRI** [comment]
The code should be available on my repository here: https://github.com/TRI-ML/vidar/blob/main/vidar/utils/flow.py
Let me know if it works for you!

**Comment by nauti16** [comment]
Dear VitorGuizilini-TRI,
I have reviewed your code and confirmed that the extrinsic pose needs to be inverted. 
I missed that in VKITTI2.0 dataset. Thank you very much!!!
